### PR TITLE
Add mixer based hda topology for MTL

### PR DESCRIPTION
**Pull request by fredoh9**
This is cavs-passthrough-hda-4ch-mtl.tplg, which is already in the CMakLists.txt
cavs-passthrough-hda-4ch-mtl

This is new topology from this PR, cavs-mixin-mixout-hda-4ch-mtl.tplg
I tested with Realtek AIOC40, ALC256 HDA codec.
cavs-mixin-mixout-hda-4ch-mtl

### CODE REVIEW

**Comment by fredoh9** [comment]
I allow to run SOF CI once to see making sure everything is fine.
Then put 'skip ci' as this topology is not being used by SOF CI yet

**Comment by lgirdwood** [comment]
@bardliao @ranj063 pls take a look thanks.

File: tools/topology/topology2/cavs/CMakeLists.txt [hunk 1]
```diff
@@ -8,22 +8,24 @@ set(TPLGS
...
+HDA_CONFIG=mix,NUM_DMICS=2,PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt.bin"
+"cavs-passthrough-hdmi\;cavs-mixin-mixout-hda-4ch-tgl\;\
 HDA_CONFIG=mix,NUM_DMICS=4,PLATFORM=cnl,PDM1_MIC_A_ENABLE=1,PDM1_MIC_B_ENABLE=1,\
 PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt.bin"
+# MTL: CAVS HDA topology with passthrough pipelines for HDMI
 "cavs-passthrough-hdmi\;cavs-passthrough-hda-4ch-mtl\;PLATFORM=mtl,\
 NUM_DMICS=4,PDM1_MIC_A_ENABLE=1,PDM1_MIC_B_ENABLE=1,PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt-mtl.bin"
+# MTL: CAVS HDA topology with mixer-based pipelines for HDA and passthrough pipelines for HDMI
+"cavs-passthrough-hdmi\;cavs-mixin-mixout-hda-4ch-mtl\;PLATFORM=mtl,\
+HDA_CONFIG=mix,NUM_DMICS=4,PDM1_MIC_A_ENABLE=1,PDM1_MIC_B_ENABLE=1,PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt-mtl.bin"
```
Comment (bardliao) [inline tools/topology/topology2/cavs/CMakeLists.txt:28]: We probably want to use a different nhlt bin name. "nhlt-mtl.bin" is used by cavs-passthrough-hda-4ch-mtl

Comment (fredoh9) [reply tools/topology/topology2/cavs/CMakeLists.txt]: Changed to nhlt-mtl-hda-mix.bin

**Review by bardliao — APPROVED** [review]

File: tools/topology/topology2/cavs/CMakeLists.txt [hunk 2]
```diff
@@ -23,6 +23,9 @@ PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt.bin"
 # MTL: CAVS HDA topology with passthrough pipelines for HDMI
 "cavs-passthrough-hdmi\;cavs-passthrough-hda-4ch-mtl\;PLATFORM=mtl,\
 NUM_DMICS=4,PDM1_MIC_A_ENABLE=1,PDM1_MIC_B_ENABLE=1,PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt-mtl.bin"
+# MTL: CAVS HDA topology with mixer-based pipelines for HDA and passthrough pipelines for HDMI
```
[the remark this answers is not in the document]
Comment (fredoh9) [reply tools/topology/topology2/cavs/CMakeLists.txt:26]: yes, I was also wondering what we can do with passthrough HDMI pipelines only. Will remove it and update the pr again

**Comment by fredoh9** [comment]
SOFCI TEST

**Comment by fredoh9** [comment]
@lgirdwood everything is GREEN

**Review by ranj063 — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**